### ISSUE TITLE
EvalException are not caught by try()

### ISSUE THREAD

**Issue by akbertram** [bedatadriven/renjin#134]
Given the script:
```test.R
p <- quote(foo(x))
try({ p[NA] })
print(1)
```
When you run `renjin -f test.R`, the process exits abrubtly despite the try statement. 

```
alex@alex-laptop76:~/dev/renjin$ renjin -f /tmp/test.R 
Java home = /usr/lib/jvm/java-7-openjdk-amd64
Using local repository: /home/alex/.m2/repository
Renjin 0.8.1580
Copyright (C) 2013 The R Foundation for Statistical Computing
Copyright (C) 2014 BeDataDriven
> p <- quote(foo(x))
> try({ p[NA] })
ERROR: position (-2147483648) must not be negativejava.lang.IndexOutOfBoundsException: position (-2147483648) must not be negative
        at com.google.common.collect.Iterators.checkNonnegative(Iterators.java:822)
        at com.google.common.collect.Iterators.get(Iterators.java:810)
        at org.renjin.sexp.PairList$Node.getNode(PairList.java:276)
        at org.renjin.primitives.subset.Subsetting.getCallSubset(Subsetting.java:370)
        at org.renjin.primitives.subset.Subsetting.getSubset(Subsetting.java:351)
        at org.renjin.primitives.R$primitive$$bracket.apply(R$primitive$$bracket.java:63)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.sexp.Promise.doEval(Promise.java:90)
        at org.renjin.sexp.Promise.force(Promise.java:83)
        at org.renjin.eval.Context.evaluate(Context.java:196)
        at org.renjin.eval.Context.evaluateSymbol(Context.java:240)
        at org.renjin.eval.Context.evaluate(Context.java:190)
        at org.renjin.sexp.Promise.doEval(Promise.java:90)
        at org.renjin.sexp.Promise.force(Promise.java:83)
        at org.renjin.eval.Context.evaluate(Context.java:196)
        at org.renjin.eval.Context.evaluateSymbol(Context.java:240)
        at org.renjin.eval.Context.evaluate(Context.java:190)
        at org.renjin.sexp.Promise.doEval(Promise.java:90)
        at org.renjin.sexp.Promise.force(Promise.java:83)
        at org.renjin.eval.Context.evaluate(Context.java:196)
        at org.renjin.eval.Context.evaluateSymbol(Context.java:240)
        at org.renjin.eval.Context.evaluate(Context.java:190)
        at org.renjin.sexp.Promise.doEval(Promise.java:90)
        at org.renjin.sexp.Promise.force(Promise.java:83)
        at org.renjin.eval.Context.evaluate(Context.java:196)
        at org.renjin.eval.Context.evaluateSymbol(Context.java:240)
        at org.renjin.eval.Context.evaluate(Context.java:190)
        at org.renjin.primitives.special.ReturnFunction.apply(ReturnFunction.java:41)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.sexp.Promise.doEval(Promise.java:90)
        at org.renjin.sexp.Promise.force(Promise.java:83)
        at org.renjin.eval.Context.evaluate(Context.java:196)
        at org.renjin.eval.Context.evaluateSymbol(Context.java:240)
        at org.renjin.eval.Context.evaluate(Context.java:190)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.eval.Context.evaluate(Context.java:162)
        at org.renjin.sexp.Closure.doApply(Closure.java:108)
        at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:76)
        at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:61)
        at org.renjin.sexp.Closure.apply(Closure.java:82)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.AssignLeftFunction.assignLeft(AssignLeftFunction.java:60)
        at org.renjin.primitives.special.AssignLeftFunction.apply(AssignLeftFunction.java:44)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.eval.Context.evaluate(Context.java:162)
        at org.renjin.sexp.Closure.doApply(Closure.java:108)
        at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:76)
        at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:61)
        at org.renjin.sexp.Closure.apply(Closure.java:82)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.IfFunction.apply(IfFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.IfFunction.apply(IfFunction.java:50)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.eval.Context.evaluate(Context.java:162)
        at org.renjin.sexp.Closure.doApply(Closure.java:108)
        at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:76)
        at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:61)
        at org.renjin.sexp.Closure.apply(Closure.java:82)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.eval.Context.evaluate(Context.java:162)
        at org.renjin.sexp.Closure.doApply(Closure.java:108)
        at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:76)
        at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:61)
        at org.renjin.sexp.Closure.apply(Closure.java:82)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.primitives.special.BeginFunction.apply(BeginFunction.java:46)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.eval.Context.evaluate(Context.java:162)
        at org.renjin.sexp.Closure.doApply(Closure.java:108)
        at org.renjin.eval.ClosureDispatcher.apply(ClosureDispatcher.java:76)
        at org.renjin.eval.ClosureDispatcher.applyClosure(ClosureDispatcher.java:61)
        at org.renjin.sexp.Closure.apply(Closure.java:82)
        at org.renjin.eval.Context.evaluateCall(Context.java:262)
        at org.renjin.eval.Context.evaluate(Context.java:194)
        at org.renjin.repl.JlineRepl.readExpression(JlineRepl.java:169)
        at org.renjin.repl.JlineRepl.run(JlineRepl.java:101)
        at org.renjin.cli.Main.evaluate(Main.java:137)
        at org.renjin.cli.Main.evaluateFile(Main.java:122)
        at org.renjin.cli.Main.run(Main.java:97)
        at org.renjin.cli.Main.main(Main.java:71)
Execution halted
```

**Comment by mjkallen** [comment]
Being able to handle Java exceptions in the R functions inside Renjin extension packages is definitely a missing feature.

**Comment by akbertram** [comment]
The R code above actually *shouldn't* throw an Exception :-)
That is fixed in 3494e41f36b204136d67c2a7703f72be0d85add1.
